res/controllers/Numark-Mixstream-Pro.scripts.js

```diff
@@ -0,0 +1,976 @@
+                        });
+                    } else {
+                        engine.setValue(group, "loop_out", 1);
+                        engine.setValue(group, `hotcue_${loopNum}_activateloop`, 1);
+                        engine.setValue(group, "loop_in", 0);
+                        // Stop LED flashing and turn LED solid on
+                        engine.stopTimer(deckState.padButton.blinktimer);
+                        deckState.padButton.blinktimer = 0;
+                        ledOn(deckState.midiStatus, control);
+                        // Small delay before activating loop to ensure Mixxx has processed loop_out
+                        engine.beginTimer(100, function() {
+                            engine.setValue(group, `hotcue_${loopNum}_activateloop`, 1);
+                        }, 1);
+                    }
+                }
+            } else if (value === 0) {
+                engine.setValue(group, `hotcue_${loopNum}_activateloop`, 0);
+            }
+        }
+    }
+    // AUTOLOOP MODE
+    if (value === 127 && deckState.padModes.autoloop) {
+        const loopSize = deckState.padModes.autoloop === 1 ? config.autoloopBank1 : config.autoloopBank2;
+        engine.setValue(group, `beatloop_${loopSize}_toggle`, true);
+
+        // Send LED feedback
+        for (let i = 1; i <= 4; i++) {
+            if (i === padNumber && engine.getValue(group, `beatloop_${loopSize}_enabled`)) {
+                ledOn(status, padIndexToCC(i));
+            } else {
+                ledDim(status, padIndexToCC(i));
+            }
+        }
+    }
+
+    // BEATLOOPROLL MODE
+    if (deckState.padModes.roll) {
+        if (value === 127) {
+            const loopSize = deckState.padModes.roll === 1 ? config.beatloopRollBank1 : config.beatloopRollBank2;
+            engine.setValue(group, "beatloop_size", loopSize);
+            if (!engine.getValue(group, "loop_enabled")) {
+                engine.setValue(group, "beatlooproll_activate", true);
+            }
+            // Send LED feedback
+            ledOn(status, control);
+        } else if (value === 0) {
+            engine.setValue(group, "beatlooproll_activate", false);
+            ledDim(status, control);
+            // Restore slip mode after beatloop roll deactivates (Mixxx disables it automatically)
+            if (deckState.jogWheel.scratchMode === SCRATCH_MODE.smart) {
+                engine.beginTimer(10, function() {
+                    engine.setValue(group, "slip_enabled", true);
+                    ledOn(status, LED_ADDR.SLIP);
+                }, 1);
+            }
+        }
+    }
+
+    // SAMPLER MODE
+    if (value === 127 && deckState.padModes.sampler) {
+        const sample = padIndex(padNumber, deckState.padModes.sampler, "sampler");
+        if (deckState.shift) {
+            engine.setValue(`[Sampler${sample}]`, "stop", 1);
+        } else {
+            engine.setValue(`[Sampler${sample}]`, "cue_gotoandplay", 1);
+        }
+    }
```

Comment:
You could also implement this with switch/case, will likely be more compact then